native/src/main/java/io/ballerina/stdlib/http/api/HttpCallableUnitCallback.java

```diff
@@ -78,8 +80,17 @@ public void notifyFailure(BError result) {
                 sendFailureResponse(result);
             }
         };
-        runtime.invokeMethodAsync(caller, "returnResponse", null, ModuleUtils.getNotifySuccessMetaData(),
-                                  returnCallback, paramFeed);
+        ObjectType callerType = caller.getType();
+        String remoteMethod = "returnResponse";
+        if (callerType.isIsolated() && callerType.isIsolated(remoteMethod)) {
```

Comment:
Here also the same

Reply:
Fixed with [4d461ab74ee46c7a163d480ff8827652e794cf51](https://github.com/ballerina-platform/module-ballerina-http/pull/765/commits/4d461ab74ee46c7a163d480ff8827652e794cf51)